### ISSUE TITLE
[PAID] [$2000] Message actions popover not showing after open app download links page and go back.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14131]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open any chat room.
2. Right click on any message to open actions popover. (Popover will be display)
3. Go to Settings => About => App download links.
4. Dismiss/close modal on outside click.
5. Try to right click on any message to open actions popover/modal.

## Expected Result:
Should show actions popover of message on right click - App download links page shouldn’t disturb it.

## Actual Result:
Actions of message popover not showing after we open App download links and close it.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.50-13
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/211237474-c9bbefbc-405e-42e7-bd98-bae14c81718f.mov

https://user-images.githubusercontent.com/43996225/211237503-46768d82-ef8d-4481-bfcd-01f7d149da4e.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @jatinsonijs 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673182112110259

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013b41ce2fcee99410</li>
        <li>Upwork Job ID: 1612963362142433280</li>
        <li>Last Price Increase: 2023-01-18</li>
    </ul>
</details>

**Comment by strepanier03** [comment]
### Bug0 Triage Checklist
_Note: see [this SO](https://stackoverflow.com/c/expensify/questions/14418) for more information._
- [ ] The "bug" is actually a bug 
- [ ] The bug is not a duplicate report of an existing GH.
  - If it is, close the GH and add any novel details to the original GH instead 
- [ ] The bug is reproducible, following the reproduction steps.
  - [ ] If the GH doesn’t have steps to reliably reproduce the bug and you figure it out, then please add them.
  - [ ] If you’re unable to reproduce the bug, add the `Needs reproduction` label. 
  - [ ] Comment on the issue outlining the steps you took to try to reproduce the bug, your results and tag the issue reporter and the Applause QA member who created the issue. Ask them to clarify reproduction steps and/or to check the reproduction steps again. Close issue. 
- [ ] The GH template is filled out as fully as possible
  - The GH body and title are clear (ie. could an external contributor understand it and work on it?) 
- [ ] The GH was created by an Expensify employee or a QA tester 
- [ ] If the bug is an OldDot issue, you should decide whether it is widespread enough to justify fixing or it is better to wait for reunification. If it’s better to wait, close the GH & provide this justification 
- [ ] If there's a link to Slack, check the discussion to see if we decided not to fix it 
- [ ] Decide if the GH should be resolved by an External contributor or Internal engineer, add the appropriate label

**Comment by strepanier03** [comment]
I am able to recreate this following the steps in the OG comment. 

![2023-01-10_15-58-26 (1)](https://user-images.githubusercontent.com/10925636/211687103-7073da03-00fc-4028-b72a-637756f2d21d.gif)

**Comment by allroundexperts** [comment]
## Proposal
This issue was introduced as a result of this PR: https://github.com/Expensify/App/issues/11893. The root cause of this is the use of global reference (exported from `ReportActionContextMenu`) to the `PopoverReportActionContextMenu` component which gets mutated by any other component using `ReportActionContextMenu`.

**Comment by tienifr** [comment]
### Proposal

**Problem**
We're using `ReportActionContextMenu.contextMenuRef` in 2 places in `src/pages/home/report/ReportActionsView.js` and `src/pages/settings/AppDownloadLinks.js`. When the App Download Link sidebar is opened, the `contextMenuRef` is set to the component in that page, when we dismiss the sidebar, the ref is set to null/undefined and not reset to the component in `ReportActionsView.js`, so the context menu could not be opened.

**Solution**
The `<PopoverReportActionContextMenu` in `src/pages/home/report/ReportActionsView.js` is redundant because it can already use the `contextMenuRef` defined in`ReportActionsView.js`,  so we should remove it.

I think we shouldn't assign a single ref to 2 different components elsewhere as well.

<img width="914" alt="Screenshot 2023-01-11 at 08 51 03" src="https://user-images.githubusercontent.com/113963320/211699433-6988d689-b574-4570-a0bc-89706d37d2d4.png">

**Comment by tienifr** [comment]
It's working fine after the fix

[Demo fix.mov.zip](https://github.com/Expensify/App/files/10388179/Demo.fix.mov.zip)
The video is too big for Github (>10MB) so I had to zip it before uploading

**Comment by tienifr** [comment]
# Proposal

## Problem

We're using ReportActionContextMenu.contextMenuRef in 2 places in src/pages/home/report/ReportActionsView.js and src/pages/settings/AppDownloadLinks.js. When the App Download Link sidebar is opened, the contextMenuRef is set to the component in that page, when we dismiss the sidebar, the ref is set to null/undefined and not reset to the component in ReportActionsView.js, so the context menu could not be opened.

## Solution

I'll assign the separate ref for each page (ReportActionViews and AppDownloadLink) by creating 2 refs: reportActionRef, appDownloadLinkRef.

I create the new function `getContextMenuRef` to get ref base on the pageName (return reportActionRef by default)

I'll also add new argument `page` to each function in `ReportActionContextMenu` to get the appropriate ref

```diff

diff --git a/src/pages/home/report/ContextMenu/ReportActionContextMenu.js b/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
index 6d14773a8f..a9b90c431c 100644
--- a/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
+++ b/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
@@ -1,6 +1,18 @@
 import React from 'react';
 
-const contextMenuRef = React.createRef();
+const reportActionRef = React.createRef();
+const appDownloadLinkRef = React.createRef();
+
+/**
+ * Get ref base on the page
+ * @param {String} page [ReportActionsView, AppDownloadLink]
+ */
+function getContextMenuRef(page){
+    if(page == 'AppDownloadLink'){
+        return appDownloadLinkRef
+    }
+    return reportActionRef
+}
 
 /**
  * Show the ReportActionContextMenu modal popover.
@@ -14,6 +26,7 @@ const contextMenuRef = React.createRef();
  * @param {String} draftMessage - ReportAction Draftmessage
  * @param {Function} [onShow=() => {}] - Run a callback when Menu is shown
  * @param {Function} [onHide=() => {}] - Run a callback when Menu is hidden
+ * @param {String} page [ReportActionsView, AppDownloadLink]
  */
 function showContextMenu(
     type,
@@ -25,7 +38,9 @@ function showContextMenu(
     draftMessage = '',
     onShow = () => {},
     onHide = () => {},
+    page = 'ReportActionsView'
 ) {
+    const contextMenuRef = getContextMenuRef(page);
     if (!contextMenuRef.current) {
         return;
     }
@@ -47,8 +62,10 @@ function showContextMenu(
  * Hides the popover menu with an optional delay
  * @param {Boolean} shouldDelay - whether the menu should close after a delay
  * @param {Function} [onHideCallback=() => {}] - Callback to be called after Context Menu is completely hidden
+ * @param {String} page [Report, AppDownloadLink]
  */
-function hideContextMenu(shouldDelay, onHideCallback = () => {}) {
+function hideContextMenu(shouldDelay, onHideCallback = () => {}, page) {
+    const contextMenuRef = getContextMenuRef(page);
     if (!contextMenuRef.current) {
         return;
     }
@@ -71,7 +88,8 @@ function hideContextMenu(shouldDelay, onHideCallback = () => {}) {
     }, 800);
 }
 
-function hideDeleteModal() {
+function hideDeleteModal(page) {
+    const contextMenuRef = getContextMenuRef(page);
     if (!contextMenuRef.current) {
         return;
     }
@@ -86,7 +104,8 @@ function hideDeleteModal() {
  * @param {Function} [onConfirm]
  * @param {Function} [onCancel]
  */
-function showDeleteModal(reportID, reportAction, shouldSetModalVisibility, onConfirm, onCancel) {
+function showDeleteModal(reportID, reportAction, shouldSetModalVisibility, onConfirm, onCancel, page) {
+    const contextMenuRef = getContextMenuRef(page);
     if (!contextMenuRef.current) {
         return;
     }
@@ -99,7 +118,8 @@ function showDeleteModal(reportID, reportAction, shouldSetModalVisibility, onCon
  * @param {Number|String} actionID
  * @return {Boolean}
  */
-function isActiveReportAction(actionID) {
+function isActiveReportAction(actionID, page) {
+    const contextMenuRef = getContextMenuRef(page);
     if (!contextMenuRef.current) {
         return;
     }
@@ -107,7 +127,7 @@ function isActiveReportAction(actionID) {
 }
 
 export {
-    contextMenuRef,
+    getContextMenuRef,
     showContextMenu,
     hideContextMenu,
     isActiveReportAction,
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 25aba06a5f..c000141d2c 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -372,7 +372,7 @@ class ReportActionsView extends React.Component {
                             newMarkerReportActionID={this.state.newMarkerReportActionID}
                         />
                         <PopoverReportActionContextMenu
-                            ref={ReportActionContextMenu.contextMenuRef}
+                            ref={ReportActionContextMenu.getContextMenuRef()}
                             isArchivedRoom={ReportUtils.isArchivedRoom(this.props.report)}
                             isChronosReport={ReportUtils.chatIncludesChronos(this.props.report)}
                         />
diff --git a/src/pages/settings/AppDownloadLinks.js b/src/pages/settings/AppDownloadLinks.js
index 2738d18441..6fba4e3b90 100644
--- a/src/pages/settings/AppDownloadLinks.js
+++ b/src/pages/settings/AppDownloadLinks.js
@@ -69,6 +69,12 @@ const AppDownloadLinksPage = (props) => {
             event,
             selection,
             popoverAnchor,
+            '0',
+            {},
+            '',
+            () => {},
+            () => {},
+            'AppDownloadLink'
         );
     };
 
@@ -82,7 +88,7 @@ const AppDownloadLinksPage = (props) => {
             />
             <ScrollView style={[styles.mt5]}>
                 <PopoverReportActionContextMenu
-                    ref={ReportActionContextMenu.contextMenuRef}
+                    ref={ReportActionContextMenu.getContextMenuRef('AppDownloadLink')}
                 />
                 {_.map(menuItems, item => (
                     <PressableWithSecondaryInteraction

```

## Result

https://user-images.githubusercontent.com/113963320/211710863-c8b366a8-b592-4221-ae79-b7567703b40b.mov

**Comment by Someone-here** [comment]
### Update

I noticed that this again introduces the problem of not having the url copying menu as in the issue #11893 on mWeb and this is because on desktop web, the ReportScreen is mounted in the background when the appdownloadLinks page is shown. So it works properly there. The only problem is the report screen is not mounted on mWeb when opening settings. So again the solution for this is to check if the device `issmallWidthScreen` and if it is, only then we should add the `PopoverContextMenu` with the ref. 

**Solution**

<img width="1257" alt="Screenshot 2023-01-11 at 8 59 26 AM" src="https://user-images.githubusercontent.com/77237602/211711269-4bc9f7b5-7f36-4e6d-8b33-e245b2cbf6fb.png">

This doesn't introduce the same issue because the appDownloadLinks is unmounted after the drawer is closed, so it will not effect the ReportScreen.
**This solution provides the least complexity and doesn't need to create additional refs that can be difficult to manage as in @tienifr second proposal.**

**After Fix**
(Sorry for the quality, it is because of github file limit)

https://user-images.githubusercontent.com/77237602/211712118-c3141142-a22c-4e08-a9c6-9a48a7d283e2.mov

I also tested this on a real device and it works as expected.

**Comment by tienifr** [comment]
@Someone-here Your proposal doesn't work correctly if you go back the report that you visited just before

**Comment by tienifr** [comment]
# Proposal 2

On large screen we can remove `PopoverReportActionContextMenu` in `AppDownloadLink` because ReportActionsView always renders even when user visit appDownloadLink page first => ReportActionContextMenu.contextMenuRef is assigned

On small screen we have to use  `PopoverReportActionContextMenu` since when user visit appDownloadLink page first, ReportActionsView is not rendered => contextMenuRef is empty

## Solution

I just use `PopoverReportActionContextMenu` in `AppDownloadLink` on mobile mode

I preserve the stale ref in `ReportActionsView` and reassign contextMenuRef when it's empty

```diff

diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 25aba06a5f..4f9397df53 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -81,6 +81,7 @@ class ReportActionsView extends React.Component {
         this.recordTimeToMeasureItemLayout = this.recordTimeToMeasureItemLayout.bind(this);
         this.scrollToBottomAndMarkReportAsRead = this.scrollToBottomAndMarkReportAsRead.bind(this);
         this.openReportIfNecessary = this.openReportIfNecessary.bind(this);
+        this.contextMenuRef={current:null};
     }
 
     componentDidMount() {
@@ -130,6 +131,11 @@ class ReportActionsView extends React.Component {
     }
 
     shouldComponentUpdate(nextProps, nextState) {
+        if(!ReportActionContextMenu.contextMenuRef.current){
+            ReportActionContextMenu.contextMenuRef.current = this.contextMenuRef.current
+        }else if(!this.contextMenuRef.current){
+            this.contextMenuRef.current=ReportActionContextMenu.contextMenuRef.current
+        }
         if (!_.isEqual(nextProps.reportActions, this.props.reportActions)) {
             this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(nextProps.reportActions
);
             this.mostRecentIOUReportActionID = ReportActionsUtils.getMostRecentIOUReportActionID(nextProps.repo
rtActions);
diff --git a/src/pages/settings/AppDownloadLinks.js b/src/pages/settings/AppDownloadLinks.js
index 2738d18441..ffdc1becea 100644
--- a/src/pages/settings/AppDownloadLinks.js
+++ b/src/pages/settings/AppDownloadLinks.js
@@ -81,9 +81,9 @@ const AppDownloadLinksPage = (props) => {
                 onCloseButtonPress={() => Navigation.dismissModal(true)}
             />
             <ScrollView style={[styles.mt5]}>
-                <PopoverReportActionContextMenu
+                {props.isSmallScreenWidth && <PopoverReportActionContextMenu
                     ref={ReportActionContextMenu.contextMenuRef}
-                />
+                />}
                 {_.map(menuItems, item => (
                     <PressableWithSecondaryInteraction
                         key={item.translationKey}

```

**Comment by tienifr** [comment]
# Proposal 3

So I was able to find another solution that I believe will address the root cause of this. The root cause is mentioned in the original proposal [here](https://github.com/Expensify/App/issues/14131#issuecomment-1378196147). We're using `ReportActionContextMenu.showContextMenu` as a generic method that's used app-wide. So if we still have 2 (or more) different `PopoverReportActionContextMenu` in different places in the app, it will surely cause some race conditions since by right, the same ref should not be associated with 2 different components.
We should consider `PopoverReportActionContextMenu` as an app-wide component instead so that it can be reused in different places in the app (both the ReportScreen and AppDownloadLinks as of now). This can be done by escalating `PopoverReportActionContextMenu` up to the `src/libs/Navigation/AppNavigator/MainDrawerNavigator.js`.

Here're the changes:
```diff
diff --git a/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js b/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
index f08cd7959e..f86ad146cd 100644
--- a/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
+++ b/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
@@ -14,6 +14,8 @@ import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import BaseDrawerNavigator from './BaseDrawerNavigator';
 import * as ReportUtils from '../../ReportUtils';
 import reportPropTypes from '../../../pages/reportPropTypes';
+import * as ReportActionContextMenu from '../../../pages/home/report/ContextMenu/ReportActionContextMenu';
+import PopoverReportActionContextMenu from '../../../pages/home/report/ContextMenu/PopoverReportActionContextMenu';
 
 const propTypes = {
     /** Available reports that would be displayed in this navigator */
@@ -80,26 +82,31 @@ class MainDrawerNavigator extends Component {
         // This way routing information is updated (if needed) based on the initial report ID resolved.
         // This is usually needed after login/create account and re-launches
         return (
-            <BaseDrawerNavigator
-                drawerContent={({navigation, state}) => {
-                    // This state belongs to the drawer so it should always have the ReportScreen as it's initial (and only) route
-                    const reportIDFromRoute = lodashGet(state, ['routes', 0, 'params', 'reportID']);
-                    return (
-                        <SidebarScreen
-                            navigation={navigation}
-                            reportIDFromRoute={reportIDFromRoute}
-                        />
-                    );
-                }}
-                screens={[
-                    {
-                        name: SCREENS.REPORT,
-                        component: ReportScreen,
-                        initialParams: this.initialParams,
-                    },
-                ]}
-                isMainScreen
-            />
+            <>
+                <PopoverReportActionContextMenu
+                    ref={ReportActionContextMenu.contextMenuRef}
+                />
+                <BaseDrawerNavigator
+                    drawerContent={({navigation, state}) => {
+                        // This state belongs to the drawer so it should always have the ReportScreen as it's initial (and only) route
+                        const reportIDFromRoute = lodashGet(state, ['routes', 0, 'params', 'reportID']);
+                        return (
+                            <SidebarScreen
+                                navigation={navigation}
+                                reportIDFromRoute={reportIDFromRoute}
+                            />
+                        );
+                    }}
+                    screens={[
+                        {
+                            name: SCREENS.REPORT,
+                            component: ReportScreen,
+                            initialParams: this.initialParams,
+                        },
+                    ]}
+                    isMainScreen
+                />
+            </>
         );
     }
 }
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 25aba06a5f..5f035c807d 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -12,8 +12,6 @@ import withWindowDimensions, {windowDimensionsPropTypes} from '../../../componen
 import {withDrawerPropTypes} from '../../../components/withDrawerState';
 import * as ReportScrollManager from '../../../libs/ReportScrollManager';
 import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
-import * as ReportActionContextMenu from './ContextMenu/ReportActionContextMenu';
-import PopoverReportActionContextMenu from './ContextMenu/PopoverReportActionContextMenu';
 import Performance from '../../../libs/Performance';
 import {withNetwork} from '../../../components/OnyxProvider';
 import * as EmojiPickerAction from '../../../libs/actions/EmojiPickerAction';
@@ -371,11 +369,6 @@ class ReportActionsView extends React.Component {
                             loadMoreChats={this.loadMoreChats}
                             newMarkerReportActionID={this.state.newMarkerReportActionID}
                         />
-                        <PopoverReportActionContextMenu
-                            ref={ReportActionContextMenu.contextMenuRef}
-                            isArchivedRoom={ReportUtils.isArchivedRoom(this.props.report)}
-                            isChronosReport={ReportUtils.chatIncludesChronos(this.props.report)}
-                        />
                     </>
                 )}
                 <EmojiPicker ref={EmojiPickerAction.emojiPickerRef} />
diff --git a/src/pages/settings/AppDownloadLinks.js b/src/pages/settings/AppDownloadLinks.js
index 2738d18441..9b47bdb644 100644
--- a/src/pages/settings/AppDownloadLinks.js
+++ b/src/pages/settings/AppDownloadLinks.js
@@ -17,7 +17,6 @@ import withWindowDimensions, {windowDimensionsPropTypes} from '../../components/
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 import * as ReportActionContextMenu from '../home/report/ContextMenu/ReportActionContextMenu';
 import * as ContextMenuActions from '../home/report/ContextMenu/ContextMenuActions';
-import PopoverReportActionContextMenu from '../home/report/ContextMenu/PopoverReportActionContextMenu';
 
 const propTypes = {
     ...withLocalizePropTypes,
@@ -81,9 +80,6 @@ const AppDownloadLinksPage = (props) => {
                 onCloseButtonPress={() => Navigation.dismissModal(true)}
             />
             <ScrollView style={[styles.mt5]}>
-                <PopoverReportActionContextMenu
-                    ref={ReportActionContextMenu.contextMenuRef}
-                />
                 {_.map(menuItems, item => (
                     <PressableWithSecondaryInteraction
                         key={item.translationKey}

```

Of course there'll be some further small changes required to pass the 
```diff
-                            isArchivedRoom={ReportUtils.isArchivedRoom(this.props.report)}
-                            isChronosReport={ReportUtils.chatIncludesChronos(this.props.report)}
```
into the `ReportActionContextMenu.showContextMenu` to preserve the filter logic. But the above changes should indicate the main fix that's required for this proposal.

After the fix:
https://user-images.githubusercontent.com/113963320/211764870-cc05a282-e134-4eac-bb3b-7b80643b415c.mov

**Comment by tienifr** [comment]
# Proposal 3

So I was able to find another solution that I believe will address the root cause of this. The root cause is mentioned in the original proposal [here](https://github.com/Expensify/App/issues/14131#issuecomment-1378196147). We're using `ReportActionContextMenu.showContextMenu` as a generic method that's used app-wide. So if we still have 2 (or more) different `PopoverReportActionContextMenu` in different places in the app, it will surely cause some race conditions since by right, the same ref should not be associated with 2 different components.

We should consider `PopoverReportActionContextMenu` as an app-wide component instead so that it can be reused in different places in the app (both the ReportScreen and AppDownloadLinks as of now). This can be done by escalating `PopoverReportActionContextMenu` up to the `src/libs/Navigation/AppNavigator/MainDrawerNavigator.js` (feel free to suggest a more appropriate place we should escalate it to if you think the `MainDrawerNavigator.js` might not be the best choice).

Here're the changes:
```diff
diff --git a/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js b/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
index f08cd7959e..f86ad146cd 100644
--- a/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
+++ b/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
@@ -14,6 +14,8 @@ import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import BaseDrawerNavigator from './BaseDrawerNavigator';
 import * as ReportUtils from '../../ReportUtils';
 import reportPropTypes from '../../../pages/reportPropTypes';
+import * as ReportActionContextMenu from '../../../pages/home/report/ContextMenu/ReportActionContextMenu';
+import PopoverReportActionContextMenu from '../../../pages/home/report/ContextMenu/PopoverReportActionContextMenu';
 
 const propTypes = {
     /** Available reports that would be displayed in this navigator */
@@ -80,26 +82,31 @@ class MainDrawerNavigator extends Component {
         // This way routing information is updated (if needed) based on the initial report ID resolved.
         // This is usually needed after login/create account and re-launches
         return (
-            <BaseDrawerNavigator
-                drawerContent={({navigation, state}) => {
-                    // This state belongs to the drawer so it should always have the ReportScreen as it's initial (and only) route
-                    const reportIDFromRoute = lodashGet(state, ['routes', 0, 'params', 'reportID']);
-                    return (
-                        <SidebarScreen
-                            navigation={navigation}
-                            reportIDFromRoute={reportIDFromRoute}
-                        />
-                    );
-                }}
-                screens={[
-                    {
-                        name: SCREENS.REPORT,
-                        component: ReportScreen,
-                        initialParams: this.initialParams,
-                    },
-                ]}
-                isMainScreen
-            />
+            <>
+                <PopoverReportActionContextMenu
+                    ref={ReportActionContextMenu.contextMenuRef}
+                />
+                <BaseDrawerNavigator
+                    drawerContent={({navigation, state}) => {
+                        // This state belongs to the drawer so it should always have the ReportScreen as it's initial (and only) route
+                        const reportIDFromRoute = lodashGet(state, ['routes', 0, 'params', 'reportID']);
+                        return (
+                            <SidebarScreen
+                                navigation={navigation}
+                                reportIDFromRoute={reportIDFromRoute}
+                            />
+                        );
+                    }}
+                    screens={[
+                        {
+                            name: SCREENS.REPORT,
+                            component: ReportScreen,
+                            initialParams: this.initialParams,
+                        },
+                    ]}
+                    isMainScreen
+                />
+            </>
         );
     }
 }
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 25aba06a5f..5f035c807d 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -12,8 +12,6 @@ import withWindowDimensions, {windowDimensionsPropTypes} from '../../../componen
 import {withDrawerPropTypes} from '../../../components/withDrawerState';
 import * as ReportScrollManager from '../../../libs/ReportScrollManager';
 import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
-import * as ReportActionContextMenu from './ContextMenu/ReportActionContextMenu';
-import PopoverReportActionContextMenu from './ContextMenu/PopoverReportActionContextMenu';
 import Performance from '../../../libs/Performance';
 import {withNetwork} from '../../../components/OnyxProvider';
 import * as EmojiPickerAction from '../../../libs/actions/EmojiPickerAction';
@@ -371,11 +369,6 @@ class ReportActionsView extends React.Component {
                             loadMoreChats={this.loadMoreChats}
                             newMarkerReportActionID={this.state.newMarkerReportActionID}
                         />
-                        <PopoverReportActionContextMenu
-                            ref={ReportActionContextMenu.contextMenuRef}
-                            isArchivedRoom={ReportUtils.isArchivedRoom(this.props.report)}
-                            isChronosReport={ReportUtils.chatIncludesChronos(this.props.report)}
-                        />
                     </>
                 )}
                 <EmojiPicker ref={EmojiPickerAction.emojiPickerRef} />
diff --git a/src/pages/settings/AppDownloadLinks.js b/src/pages/settings/AppDownloadLinks.js
index 2738d18441..9b47bdb644 100644
--- a/src/pages/settings/AppDownloadLinks.js
+++ b/src/pages/settings/AppDownloadLinks.js
@@ -17,7 +17,6 @@ import withWindowDimensions, {windowDimensionsPropTypes} from '../../components/
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 import * as ReportActionContextMenu from '../home/report/ContextMenu/ReportActionContextMenu';
 import * as ContextMenuActions from '../home/report/ContextMenu/ContextMenuActions';
-import PopoverReportActionContextMenu from '../home/report/ContextMenu/PopoverReportActionContextMenu';
 
 const propTypes = {
     ...withLocalizePropTypes,
@@ -81,9 +80,6 @@ const AppDownloadLinksPage = (props) => {
                 onCloseButtonPress={() => Navigation.dismissModal(true)}
             />
             <ScrollView style={[styles.mt5]}>
-                <PopoverReportActionContextMenu
-                    ref={ReportActionContextMenu.contextMenuRef}
-                />
                 {_.map(menuItems, item => (
                     <PressableWithSecondaryInteraction
                         key={item.translationKey}

```

Of course there'll be some further small changes required to pass the 
```diff
-                            isArchivedRoom={ReportUtils.isArchivedRoom(this.props.report)}
-                            isChronosReport={ReportUtils.chatIncludesChronos(this.props.report)}
```
into the `ReportActionContextMenu.showContextMenu` to preserve the filter logic (I'll add these shortly). But the above changes should indicate the main fix that's required for this proposal.

After the fix:
https://user-images.githubusercontent.com/113963320/211764870-cc05a282-e134-4eac-bb3b-7b80643b415c.mov

**Comment by tienifr** [comment]
Continue from https://github.com/Expensify/App/issues/14131#issuecomment-1378451746

> Of course there'll be some further small changes required to pass the isArchivedRoom, isChronosReport into the ReportActionContextMenu.showContextMenu to preserve the filter logic (I'll add these shortly). But the above changes should indicate the main fix that's required for this proposal.

```diff
diff --git a/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js b/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js
index 2f174ce5cc..770844edc3 100644
--- a/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js
+++ b/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js
@@ -3,7 +3,6 @@ import {
     Dimensions,
 } from 'react-native';
 import _ from 'underscore';
-import PropTypes from 'prop-types';
 import lodashGet from 'lodash/get';
 import * as Report from '../../../../libs/actions/Report';
 import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
@@ -12,20 +11,9 @@ import BaseReportActionContextMenu from './BaseReportActionContextMenu';
 import ConfirmModal from '../../../../components/ConfirmModal';
 
 const propTypes = {
-    /** Flag to check if the chat participant is Chronos */
-    isChronosReport: PropTypes.bool,
-
-    /** Whether the provided report is an archived room */
-    isArchivedRoom: PropTypes.bool,
-
     ...withLocalizePropTypes,
 };
 
-const defaultProps = {
-    isChronosReport: false,
-    isArchivedRoom: false,
-};
-
 class PopoverReportActionContextMenu extends React.Component {
     constructor(props) {
         super(props);
@@ -48,6 +36,8 @@ class PopoverReportActionContextMenu extends React.Component {
                 horizontal: 0,
                 vertical: 0,
             },
+            isArchivedRoom: false,
+            isChronosReport: false,
         };
         this.onPopoverShow = () => {};
         this.onPopoverHide = () => {};
@@ -126,6 +116,8 @@ class PopoverReportActionContextMenu extends React.Component {
      * @param {String} draftMessage - ReportAction Draftmessage
      * @param {Function} [onShow] - Run a callback when Menu is shown
      * @param {Function} [onHide] - Run a callback when Menu is hidden
+     * @param {Boolean} isArchivedRoom - isArchivedRoom
+     * @param {Boolean} isChronosReport - isChronosReport
      */
     showContextMenu(
         type,
@@ -137,6 +129,8 @@ class PopoverReportActionContextMenu extends React.Component {
         draftMessage,
         onShow = () => {},
         onHide = () => {},
+        isArchivedRoom,
+        isChronosReport,
     ) {
         const nativeEvent = event.nativeEvent || {};
         this.contextMenuAnchor = contextMenuAnchor;
@@ -167,6 +161,8 @@ class PopoverReportActionContextMenu extends React.Component {
                 selection,
                 isPopoverVisible: true,
                 reportActionDraftMessage: draftMessage,
+                isArchivedRoom,
+                isChronosReport,
             });
         });
     }
@@ -239,8 +235,8 @@ class PopoverReportActionContextMenu extends React.Component {
                 selection={this.state.selection}
                 reportID={this.state.reportID}
                 reportAction={this.state.reportAction}
-                isArchivedRoom={this.props.isArchivedRoom}
-                isChronosReport={this.props.isChronosReport}
+                isArchivedRoom={this.state.isArchivedRoom}
+                isChronosReport={this.state.isChronosReport}
                 anchor={this.contextMenuTargetNode}
             />
         );
@@ -269,6 +265,8 @@ class PopoverReportActionContextMenu extends React.Component {
             reportAction: {},
             isDeleteCommentConfirmModalVisible: false,
             shouldSetModalVisibilityForDeleteConfirmation: true,
+            isArchivedRoom: false,
+            isChronosReport: false,
         });
     }
 
@@ -336,6 +334,5 @@ class PopoverReportActionContextMenu extends React.Component {
 }
 
 PopoverReportActionContextMenu.propTypes = propTypes;
-PopoverReportActionContextMenu.defaultProps = defaultProps;
 
 export default withLocalize(PopoverReportActionContextMenu);
diff --git a/src/pages/home/report/ContextMenu/ReportActionContextMenu.js b/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
index 6d14773a8f..6acb7d5c33 100644
--- a/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
+++ b/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
@@ -14,6 +14,8 @@ const contextMenuRef = React.createRef();
  * @param {String} draftMessage - ReportAction Draftmessage
  * @param {Function} [onShow=() => {}] - Run a callback when Menu is shown
  * @param {Function} [onHide=() => {}] - Run a callback when Menu is hidden
+ * @param {Boolean} isArchivedRoom - isArchivedRoom
+ * @param {Boolean} isChronosReport - isChronosReport
  */
 function showContextMenu(
     type,
@@ -25,6 +27,8 @@ function showContextMenu(
     draftMessage = '',
     onShow = () => {},
     onHide = () => {},
+    isArchivedRoom = false,
+    isChronosReport = false,
 ) {
     if (!contextMenuRef.current) {
         return;
@@ -39,6 +43,8 @@ function showContextMenu(
         draftMessage,
         onShow,
         onHide,
+        isArchivedRoom,
+        isChronosReport,
     );
 }
 
diff --git a/src/pages/home/report/ReportActionItem.js b/src/pages/home/report/ReportActionItem.js
index be4eb73886..d36b399101 100644
--- a/src/pages/home/report/ReportActionItem.js
+++ b/src/pages/home/report/ReportActionItem.js
@@ -132,6 +132,8 @@ class ReportActionItem extends Component {
             this.props.draftMessage,
             undefined,
             this.checkIfContextMenuActive,
+            ReportUtils.isArchivedRoom(this.props.report),
+            ReportUtils.chatIncludesChronos(this.props.report),
         );
     }
 
```

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js b/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
index e48e68e329..c7444c1c4d 100644
--- a/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
+++ b/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
@@ -38,6 +38,7 @@ const BaseAnchorForCommentsOnly = (props) => {
             onSecondaryInteraction={
                 (event) => {
                     ReportActionContextMenu.showContextMenu(
+                        ReportActionContextMenu.IDS.REPORT,
                         Str.isValidEmail(props.displayName) ? ContextMenuActions.CONTEXT_MENU_TYPES.EMAIL : ContextMenuActions.CONTEXT_MENU_TYPES.LINK,
                         event,
                         props.href,
diff --git a/src/pages/home/report/ContextMenu/ReportActionContextMenu.js b/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
index 6d14773a8f..dbbf91181f 100644
--- a/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
+++ b/src/pages/home/report/ContextMenu/ReportActionContextMenu.js
@@ -1,10 +1,21 @@
 import React from 'react';
 
-const contextMenuRef = React.createRef();
+const IDS = {
+    REPORT: 0,
+    APPDOWNLOADLINKS: 1,
+};
+
+const contextMenuRefs = {};
+const setContextMenuRef = (el, id) => {
+    const newRef = React.createRef();
+    newRef.current = el;
+    contextMenuRefs[id] = newRef;
+};
 
 /**
  * Show the ReportActionContextMenu modal popover.
  *
+ * @param {string} id - the context menu ref id
  * @param {string} type - the context menu type to display [EMAIL, LINK, REPORT_ACTION]
  * @param {Object} [event] - A press event.
  * @param {String} [selection] - Copied content.
@@ -16,6 +27,7 @@ const contextMenuRef = React.createRef();
  * @param {Function} [onHide=() => {}] - Run a callback when Menu is hidden
  */
 function showContextMenu(
+    id,
     type,
     event,
     selection,
@@ -26,7 +38,8 @@ function showContextMenu(
     onShow = () => {},
     onHide = () => {},
 ) {
-    if (!contextMenuRef.current) {
+    const contextMenuRef = contextMenuRefs[id];
+    if (!contextMenuRef || !contextMenuRef.current) {
         return;
     }
     contextMenuRef.current.showContextMenu(
@@ -45,11 +58,13 @@ function showContextMenu(
 /**
  * Hide the ReportActionContextMenu modal popover.
  * Hides the popover menu with an optional delay
+ * @param {string} id - the context menu ref id
  * @param {Boolean} shouldDelay - whether the menu should close after a delay
  * @param {Function} [onHideCallback=() => {}] - Callback to be called after Context Menu is completely hidden
  */
-function hideContextMenu(shouldDelay, onHideCallback = () => {}) {
-    if (!contextMenuRef.current) {
+function hideContextMenu(id, shouldDelay, onHideCallback = () => {}) {
+    const contextMenuRef = contextMenuRefs[id];
+    if (!contextMenuRef || !contextMenuRef.current) {
         return;
     }
     if (!shouldDelay) {
@@ -71,8 +86,12 @@ function hideContextMenu(shouldDelay, onHideCallback = () => {}) {
     }, 800);
 }
 
-function hideDeleteModal() {
-    if (!contextMenuRef.current) {
+/**
+ * @param {string} id - the context menu ref id
+ */
+function hideDeleteModal(id) {
+    const contextMenuRef = contextMenuRefs[id];
+    if (!contextMenuRef || !contextMenuRef.current) {
         return;
     }
     contextMenuRef.current.hideDeleteModal();
@@ -80,14 +99,16 @@ function hideDeleteModal() {
 
 /**
  * Opens the Confirm delete action modal
+ * @param {string} id - the context menu ref id
  * @param {String} reportID
  * @param {Object} reportAction
  * @param {Boolean} [shouldSetModalVisibility]
  * @param {Function} [onConfirm]
  * @param {Function} [onCancel]
  */
-function showDeleteModal(reportID, reportAction, shouldSetModalVisibility, onConfirm, onCancel) {
-    if (!contextMenuRef.current) {
+function showDeleteModal(id, reportID, reportAction, shouldSetModalVisibility, onConfirm, onCancel) {
+    const contextMenuRef = contextMenuRefs[id];
+    if (!contextMenuRef || !contextMenuRef.current) {
         return;
     }
     contextMenuRef.current.showDeleteModal(reportID, reportAction, shouldSetModalVisibility, onConfirm, onCancel);
@@ -96,18 +117,21 @@ function showDeleteModal(reportID, reportAction, shouldSetModalVisibility, onCon
 /**
  * Whether Context Menu is active for the Report Action.
  *
+ * @param {string} id - the context menu ref id
  * @param {Number|String} actionID
  * @return {Boolean}
  */
-function isActiveReportAction(actionID) {
-    if (!contextMenuRef.current) {
+function isActiveReportAction(id, actionID) {
+    const contextMenuRef = contextMenuRefs[id];
+    if (!contextMenuRef || !contextMenuRef.current) {
         return;
     }
     return contextMenuRef.current.isActiveReportAction(actionID);
 }
 
 export {
-    contextMenuRef,
+    IDS,
+    setContextMenuRef,
     showContextMenu,
     hideContextMenu,
     isActiveReportAction,
diff --git a/src/pages/home/report/ReportActionItem.js b/src/pages/home/report/ReportActionItem.js
index be4eb73886..494807789c 100644
--- a/src/pages/home/report/ReportActionItem.js
+++ b/src/pages/home/report/ReportActionItem.js
@@ -73,7 +73,7 @@ class ReportActionItem extends Component {
         super(props);
         this.popoverAnchor = undefined;
         this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
+            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(ReportActionContextMenu.IDS.REPORT, props.action.reportActionID),
         };
         this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
         this.showPopover = this.showPopover.bind(this);
@@ -102,7 +102,7 @@ class ReportActionItem extends Component {
     }
 
     checkIfContextMenuActive() {
-        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(this.props.action.reportActionID)});
+        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(ReportActionContextMenu.IDS.REPORT, this.props.action.reportActionID)});
     }
 
     /**
@@ -123,6 +123,7 @@ class ReportActionItem extends Component {
         // is pasted, there are extra newlines in the content that we want to avoid.
         const selection = SelectionScraper.getCurrentSelection().replace(/\n/g, '');
         ReportActionContextMenu.showContextMenu(
+            ReportActionContextMenu.IDS.REPORT,
             ContextMenuActions.CONTEXT_MENU_TYPES.REPORT_ACTION,
             event,
             selection,
diff --git a/src/pages/home/report/ReportActionItemMessageEdit.js b/src/pages/home/report/ReportActionItemMessageEdit.js
index 97fffebebc..b34f074b40 100644
--- a/src/pages/home/report/ReportActionItemMessageEdit.js
+++ b/src/pages/home/report/ReportActionItemMessageEdit.js
@@ -167,6 +167,7 @@ class ReportActionItemMessageEdit extends React.Component {
         // When user tries to save the empty message, it will delete it. Prompt the user to confirm deleting.
         if (!trimmedNewDraft) {
             ReportActionContextMenu.showDeleteModal(
+                ReportActionContextMenu.IDS.REPORT,
                 this.props.reportID,
                 this.props.action,
                 false,
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 25aba06a5f..b6d1d1d350 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -372,7 +372,7 @@ class ReportActionsView extends React.Component {
                             newMarkerReportActionID={this.state.newMarkerReportActionID}
                         />
                         <PopoverReportActionContextMenu
-                            ref={ReportActionContextMenu.contextMenuRef}
+                            ref={el => ReportActionContextMenu.setContextMenuRef(el, ReportActionContextMenu.IDS.REPORT)}
                             isArchivedRoom={ReportUtils.isArchivedRoom(this.props.report)}
                             isChronosReport={ReportUtils.chatIncludesChronos(this.props.report)}
                         />
diff --git a/src/pages/settings/AppDownloadLinks.js b/src/pages/settings/AppDownloadLinks.js
index 2738d18441..9a2ee87691 100644
--- a/src/pages/settings/AppDownloadLinks.js
+++ b/src/pages/settings/AppDownloadLinks.js
@@ -65,6 +65,7 @@ const AppDownloadLinksPage = (props) => {
      */
     const showPopover = (event, selection) => {
         ReportActionContextMenu.showContextMenu(
+            ReportActionContextMenu.IDS.APPDOWNLOADLINKS,
             ContextMenuActions.CONTEXT_MENU_TYPES.LINK,
             event,
             selection,
@@ -82,7 +83,7 @@ const AppDownloadLinksPage = (props) => {
             />
             <ScrollView style={[styles.mt5]}>
                 <PopoverReportActionContextMenu
-                    ref={ReportActionContextMenu.contextMenuRef}
+                    ref={el => ReportActionContextMenu.setContextMenuRef(el, ReportActionContextMenu.IDS.APPDOWNLOADLINKS)}
                 />
                 {_.map(menuItems, item => (
                     <PressableWithSecondaryInteraction
```

### RCA
As others stated this is due to the ref being set to `null` after closing the app links modal.

### Details
In my proposal, I added the functionality to define multiple refs and use them accordingly, this is a flexible solution that allows us to manage multiple context menus as long as they have different ids.

For the ids I defined an enum `IDS` where we currently have only two values `REPORT` and `APPDOWNLOADLINKS` (we can move this to `CONST`)

This solution follows the same approach of @tienifr first proposal but it's a little extended.

**Comment by eVoloshchak** [comment]
Wow! Thank you @0xmiroslav for [this summary](https://github.com/Expensify/App/issues/14131#issuecomment-1378194984) and @tienifr, @Someone-here and @s77rt for your proposals

 I think that having multiple refs and logic to switch between them isn't the best approach, this adds a lot of complexity and we would have to alter the code if we decide to use the third `ReportActionContextMenu` in the future. 
 
 @tienifr's [proposal](https://github.com/Expensify/App/issues/14131#issuecomment-1378451746) looks promising, I agree that `PopoverReportActionContextMenu` should be an app-wide component.
 @tienifr, could you please provide us with the full code or a branch to test the changes? Does this fix both https://github.com/Expensify/App/issues/14131 and https://github.com/Expensify/App/issues/11893?

**Comment by tienifr** [comment]
@eVoloshchak 
The branch with the full code is here https://github.com/tienifr/App/tree/fork/feat/14131
thanks

**Comment by luacmartins** [comment]
I think @tienifr seems to be the best so far. @eVoloshchak did you have a chance to take a look at the full code in the branch above?

**Comment by eVoloshchak** [comment]
> I think @tienifr seems to be the best so far. @eVoloshchak did you have a chance to take a look at the full code in the branch above?

I did test @tienifr's branch, it is working well. I like that we have a global context menu instead of separate components. There are a couple of things that need to be addressed:
1. I'm not entirely sure about this, but [Expensify.js](https://github.com/Expensify/App/blob/main/src/Expensify.js) seems like a better place to put `PopoverReportActionContextMenu`, similar to how `GrowlNotification` is implemented
2. It introduces a new issue on native platforms (similar to https://github.com/Expensify/App/issues/8311): when you long press on a link inside a message, it triggers the wrong context menu 

### How it currently works (notice how the context menu is different if you press on a link)

https://user-images.githubusercontent.com/9059945/212747087-55b93947-8429-4681-a051-aab91dded808.mov

### After applying @tienifr's fix


https://user-images.githubusercontent.com/9059945/212747183-4eb2b40a-185d-4bf1-823a-812518f3396a.mov

**Comment by luacmartins** [comment]
Nice analysis @eVoloshchak!

>I'm not entirely sure about this, but [Expensify.js](https://github.com/Expensify/App/blob/main/src/Expensify.js) seems like a better place to put PopoverReportActionContextMenu, similar to how GrowlNotification is implemented

I agree

>It introduces a new issue on native platforms (similar to https://github.com/Expensify/App/issues/8311): when you long press on a link inside a message, it triggers the wrong context menu

This is not great. @tienifr we should address this.

**Comment by tienifr** [comment]
Thanks @eVoloshchak for highlighting.

> It introduces a new issue on native platforms (similar to https://github.com/Expensify/App/issues/8311): when you long press on a link inside a message, it triggers the wrong context menu

This is due to the fix for that issue (https://github.com/Expensify/App/issues/8311) was recently merged and it was not in my branch yet. I just merged main into my branch and push. You can pull and check again, the issue should be gone.

**Comment by tienifr** [comment]
@eVoloshchak 

> I'm not entirely sure about this, but [Expensify.js](https://github.com/Expensify/App/blob/main/src/Expensify.js) seems like a better place to put PopoverReportActionContextMenu, similar to how GrowlNotification is implemented

I've also pushed the changes to move the `PopoverReportActionContextMenu` to [Expensify.js](https://github.com/Expensify/App/blob/main/src/Expensify.js) as per your recommendation. And I tested it again it's still working well. thanks

**Comment by strepanier03** [comment]
Friendly bump @eVoloshchak to address @tienifr comments above. 

I'll circle back tomorrow to look for confirmation for hiring and take the necessary actions. Thanks!

**Comment by eVoloshchak** [comment]
@tienifr, awesome, thank you!
I've tested the branch again and it works well, there are no issues on native.

@luacmartins, @tienifr's [proposal](https://github.com/Expensify/App/issues/14131#issuecomment-1381298507) looks good to me (full diff [here](https://github.com/Expensify/App/compare/main...tienifr:App:fork/feat/14131))
🎀👀🎀 C+ reviewed!

**Comment by luacmartins** [comment]
I agree that @tienifr's proposal is the best one here.

**Comment by tienifr** [comment]
@eVoloshchak @luacmartins the PR is ready for review https://github.com/Expensify/App/pull/14431
thanks!

**Comment by strepanier03** [comment]
### Most recent update

PR review waiting for deploy to prod, all good here for now.

**Comment by luacmartins** [comment]
PR in prod!

**Comment by strepanier03** [comment]
@eVoloshchak and @tienifr - Please apply for the Upwork job here - https://www.upwork.com/jobs/~013b41ce2fcee99410

@jatinsonijs - I'll make the bug reporting job soon, Upwork is erroring during job creation right now so I'm blocked. This is on hold fo regressions so we have a week before payment would be eligible anyway. I'll ping you once the job is up for you to apply to.

**Comment by strepanier03** [comment]
The solution for this issue has been 🚀 deployed to production 🚀 in version 1.2.58- and is now subject to a 7-day regression period 📆. Here is the list of pull requests that resolve this issue:

https://github.com/Expensify/App/pull/14431
If no regressions arise, payment will be issued on 2023-02-01. 🎊

After the hold period, please check if any of the following need payment for this issue, and if so check them off after paying:

- [ ]  External issue reporter - @jatinsonijs - $250
- [ ]  Contributor that fixed the issue - @tienifr - $2000
- [ ]  Contributor+ that helped on the issue and/or PR - @eVoloshchak  - $2000

As a reminder, here are the bonuses/penalties that should be applied for any External issue:

Merged PR within 3 business days of assignment - 50% bonus
Merged PR more than 9 business days after assignment - 50% penalty

**Comment by strepanier03** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ]  [@eVoloshchak / @luacmartins] The PR that introduced the bug has been identified. Link to the PR:
- [ ]  [@eVoloshchak / @luacmartins] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ]  [@eVoloshchak / @luacmartins] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ]  [@strepanier03] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here - working on this part

**Comment by luacmartins** [comment]
@eVoloshchak would you mind filling the checklist above?

**Comment by eVoloshchak** [comment]
> @eVoloshchak would you mind filling the checklist above?

Sure thing, will do first three items tomorrow

**Comment by strepanier03** [comment]
@eVoloshchak - I've accepted your application and sent you the offer. @tienifr feel free to [apply](https://www.upwork.com/jobs/~013b41ce2fcee99410) as soon as you'd like and I'll keep an eye out for it so I can send you an offer. 

@jatinsonijs - You can apply for the reporting bonus job [here](https://www.upwork.com/jobs/~010cbfc0b77b4db6d1). Thanks!


----

I am working on regression test check now.

**Comment by strepanier03** [comment]
Raised reg test buddy check [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1675116991786219).

**Comment by strepanier03** [comment]
@jatinsonijs - I got turned around on that job I had you apply for, the title wasn't accurate and I thought I hired you for the wrong job so rescinded the offer. I've reopened it now, can you please reapply for https://www.upwork.com/jobs/~010cbfc0b77b4db6d1, sorry for the confusion on my part.

**Comment by eVoloshchak** [comment]
As per [this](https://github.com/Expensify/App/issues/14131#issuecomment-1378194984) comment

- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/12463

- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/issues/11893#issuecomment-1375336243

**Comment by jatinsonijs** [comment]
No problem @strepanier03, I have tried to apply again but its say already applied. I think you have to send me offer again. 

<img width="1073" alt="Screenshot 2023-02-01 at 11 34 27 AM" src="https://user-images.githubusercontent.com/114683042/215964625-96161247-566b-4294-b8a2-6ca633b4bea3.png">

**Comment by tienifr** [comment]
@strepanier03 I applied to the job, thanks!

**Comment by strepanier03** [comment]
Thank you @jatinsonijs - I'll figure this out and get your offer sent today.

Thank you @tienifr - I'll accept and send back the offer.

**Comment by strepanier03** [comment]
@eVoloshchak - Just a little reminder to finish filling out the checklist items [here](https://github.com/Expensify/App/issues/14131#issuecomment-1404409687). Or @luacmartins if you have the bandwidth.

**Comment by strepanier03** [comment]
@jatinsonijs - I had to reach out to Upwork support because it wouldn't let me send you a new offer and Upwork told me I had to make a new job for it. 

I have sent you an offer on the new job for you to accept.

**Comment by strepanier03** [comment]
@eVoloshchak - For the BZ checklist step below, did we need to raise a discussion or can I go ahead and check that one off too?

> A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

**Comment by strepanier03** [comment]
I'm still waiting on my reg test buddy [check](https://expensify.slack.com/archives/C01SKUP7QR0/p1675116991786219) and will move it to a GH once that's done.

**Comment by eVoloshchak** [comment]
> @eVoloshchak - For the BZ checklist step below, did we need to raise a discussion or can I go ahead and check that one off too?

I don't think additional discussion is needed, we have a checklist item for exactly this:
>  I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like Avatar, I verified the components using Avatar have been tested & I retested again)

**Comment by strepanier03** [comment]
Sounds great, thank you!

**Comment by strepanier03** [comment]
Everyone has been paid for this. Will close out once the reg test is done.

**Comment by strepanier03** [comment]
Bumped reg test buddy-check.

**Comment by luacmartins** [comment]
Left a comment in Slack. Those steps look good to me!

**Comment by MelvinBot** [comment]
@eVoloshchak @strepanier03 @luacmartins @tienifr this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @eVoloshchak is eligible for the Internal assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
@eVoloshchak @strepanier03 @luacmartins @tienifr this issue is now 4 weeks old and preventing us from maintaining WAQ. This should now be your highest priority. Please post below what your plan is to get a PR in review ASAP. Thanks!

**Comment by eVoloshchak** [comment]
This issue has been resolved, Melv

**Comment by strepanier03** [comment]
Waiting for reg test to be added - GH [here](https://github.com/Expensify/Expensify/issues/260896).
